src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -87,8 +87,12 @@ protected void onMessage(Context context, Intent intent) {
         // Extract the payload from the message
         Bundle extras = intent.getExtras();
         if (extras != null) {
+
+            SharedPreferences prefs = context.getSharedPreferences(PushPlugin.COM_ADOBE_PHONEGAP_PUSH, Context.MODE_PRIVATE);
+            boolean forceShow = prefs.getBoolean(FORCE_SHOW, false);
+
             // if we are in the foreground, just surface the payload, else post it to the statusbar
-            if (PushPlugin.isInForeground()) {
+            if (!forceShow && PushPlugin.isInForeground()) {
```

Comment:
Case 2 is wrong imo, the notification event should only be fired when actually clicked on the notification, as @fredgalvao said this could cause double events atm